### PR TITLE
Deconfuse boot package names.

### PR DESCRIPTION
**Pull request by hugelgupf**
We had "bootconfig", "booter", and "boot". What's what?

pkg/booter -> pkg/boot/systembooter
pkg/bootconfig -> pkg/boot/jsonboot
pkg/zimage -> pkg/boot/zimage
pkg/bzimage -> pkg/boot/bzimage

@jensdrenhaus @zaolin @insomniacslk @pmazzini can I rename pkg/bootconfig -> pkg/boot/jsonboot to disambiguate?

### CODE REVIEW

**Comment by jensdrenhaus** [comment]
When I started implementing `pkg/boot/stboot`, I thought `bootconfig` is some common denominator to all bootloaders in u-root. So I used it as well. But some time ago I encountered the `OSImage` interface in `pkg/boot.go` which looks nice and clean is now widely used in the different bootloaders in u-root. 
So I changed `stboot` to make use of it, too. (Not in master branch yet, but coming soon).
To sum it up: Do it, no concerns from my side.
Actually I wanted to ask you all, if we should drop it completely for deconfusion, since I think `localboot` is the only one left using it. But renaming makes sens of course.

**Comment by hugelgupf** [comment]
Yeah, I mean I think systemboot will continue to use it and as a wire format, it's not a bad idea for us to have some JSON format. BootConfig could either implement the OSImage interface itself, or it could have a function that returns a boot.OSImage if we want to be more compatible with the rest of u-root.
